manifests/job/present.pp

```diff
@@ -29,8 +33,19 @@
   }
 
   $jenkins_cli        = $jenkins::cli::cmd
-  $tmp_config_path    = "/tmp/${jobname}-config.xml"
-  $job_dir            = "${::jenkins::job_dir}/${jobname}"
+
+  # Add support for Folders plugin
+  if $folder {
+    $realjobname     = "${folder}/${jobname}"
+    $job_dir         = "${::jenkins::job_dir}/${folder}/jobs/${jobname}"
+    $tmp_config_path = "/tmp/${folder}/${jobname}-config.xml"
```

Comment:
The Folder plugin introduces the concept of namespaced jobs.
So stuff like:    MyProject/checkout    and   OtherProject/checkout  would be possible.

If I don't make $folder part of the temp file name you'd have duplicate   checkout-config.xml  files in /tmp.

Furthermore.  jobs who reside in a Folder have to be addressed with the folder name as it's part of the name space.